fedbiomed/common/cli.py

```diff
@@ -182,75 +182,73 @@ def initialize(self):
-            # Generate config file with the proper name, not the default name
-            if name:
-                os.environ["FBM_RESEARCHER_CONFIG_FILE"] = name
-
-            ResearcherConfig = importlib.import_module(
+           config_node = importlib.import_module("fedbiomed.node.config")
+           _component = config_node.node_component
+        elif component.lower() == "researcher":
+            os.environ["FBM_RESEARCHER_COMPONENT_ROOT"] = path
+            config_researcher = importlib.import_module(
                 "fedbiomed.researcher.config"
-            ).ResearcherConfig
-            config = ResearcherConfig(root=root, name=name, auto_generate=False)
+            )
+            _component = config_researcher.researcher_component
+        else:
+            print(f"Undefined component type {component}")
+            sys.exit(101)
 
-        return config
+        return _component
 
     def create(self, args):
         """CLI Handler for creating configuration file and assets for given component
         """
-        if args.component.lower() == "node":
-            default_config = DEFAULT_CONFIG_FILE_NAME_NODE
-        elif args.component.lower() == "researcher":
-            default_config = DEFAULT_CONFIG_FILE_NAME_RESEARCHER
-        else:
-            print(f"Undefined component type {args.component}")
-            exit(101)
-
-        name = get_config_name(args.name, args.component, default_config)
-        is_config_existing_before = is_config_existing(get_config_path(args.root, name))
 
-        config = self._create_config_instance(args.component, args.root, args.name)
-
-        # Overwrite force configuration file
-        if is_config_existing_before and args.force:
-            print("Overwriting existing configuration file")
-            config.generate(force=True)
-
-        # Use existing one (do nothing)
-        elif is_config_existing_before and not args.force:
-            if not args.use_current:
-                print(
-                    f'Configuration file "{config.path}" is alreay existing for name '
-                    "{config.name}. Please use --force option to overwrite"
-                )
-                exit(101)
-            # Generate wont do anything
-            config.generate()
+        if not args.path:
+            if args.component.lower() == "researcher":
+                component_path = os.path.join(os.getcwd(), DEFAULT_RESEARCHER_NAME)
+            else:
+                component_path = os.path.join(os.getcwd(), DEFAULT_NODE_NAME)
         else:
-            logger.info(f'Generation new configuration file "{config.name}"')
-            config.generate()
-
-    def refresh(self, args):
-        """Refreshes configuration file"""
+            component_path = args.path
+
+        # Researcher specific case ----------------------------------------------------
+        # This is a special case since researcher import
+        if args.component.lower() == "researcher":
```

Comment:
Hi @ybouilla,
We may remove `force` option. It used to make sense before because there use to be many component instantiated in the same directory. Now, each component has its own directory. If use for generation of a component we can either update config file and keep all other files in the component directory, or remove component folder and instantiate a new one. Keeping the old files will be useless since the id of the component will change. Second option does seem necessary to me since user can easily remove the folder by it self. Therefore, I propose to remove `force` option to simplify its usage and maintenance.